extensions/amp-script/0.1/amp-script.js

```diff
@@ -83,26 +107,37 @@ export class AmpScript extends AMP.BaseElement {
       }
       return [workerScript, authorScript, authorUrl];
     }),
-    // Configure callbacks.
+    // Options.
     {
-      onCreateWorker: data => {
-        dev().info(TAG, 'Create worker:', data);
-      },
-      onHydration: () => {
-        dev().info(TAG, 'Hydrated!');
-        this.element.classList.add('i-amphtml-hydrated');
-      },
-      onSendMessage: data => {
-        dev().info(TAG, 'To worker:', data);
+      status: this.status_,
+      mutationCallback: this.status_.mutationCallback.bind(this.status_),
```

Comment:
Combine with `callbacks` below?

Reply:
I was thinking about that, however, it's:

1. I think the actual API will be more complicated. I'm working on it right now on the WorkerDOM side. I think instead of a callback it will look more like a "pump", which would also allow us to buffer/queue up.
2. Were callbacks so far mainly meant to be informational only?

Reply:
1. Ack.
2. Initially yes but not the intent. `onHydration()` for example removes the partial-transparency after the worker is created.